lib/index.d.ts

```diff
@@ -116,6 +116,12 @@ export interface AttrSchema {
    * Can be used to auto-generate Attrs interface
    */
   interface?: string;
+
+  /**
+   * Explicitly require an attribute to be passed, useful when no default value can be inferred.
+   * Will log error when missing, and will fail rendering when used in template
+   */
+  required?: boolean;
 }
```

Comment:
a pass by suggestion, any chance I can ask to add a 

```
/** 
 For json values, actual tsType e.g `FooBar[]` can be noted here and be used for type & documentation auto generation by other tools.
*/
tsType?: string
```

Your call. I can do separate PR too.